### PR TITLE
Support more Jenkins pipeline name variations in groovy lexer

### PR DESCRIPTION
**Pull request by danilaml**
Support *.Jenkinsfile patters: https://github.com/search?l=&q=extension%3AJenkinsfile&type=code

### CODE REVIEW

File: lib/rouge/lexers/groovy.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class Groovy < RegexLexer
       title "Groovy"
       desc 'The Groovy programming language (http://www.groovy-lang.org/)'
       tag 'groovy'
-      filenames '*.groovy', 'Jenkinsfile'
+      filenames '*.groovy', 'Jenkinsfile', '*.Jenkinsfile'
```
Comment (tancnle) [inline lib/rouge/lexers/groovy.rb:10]: **suggestion:** Could you add a spec for this new file name, similar to [this](https://github.com/rouge-ruby/rouge/blob/master/spec/lexers/groovy_spec.rb#L11).

**Review by tancnle — COMMENTED** [review]
Thank you for your contribution @danilaml ❤️ I have added a tiny suggestion. Please let me know what you think.

File: lib/rouge/lexers/groovy.rb (hunk 1 above)
Comment (danilaml) [reply lib/rouge/lexers/groovy.rb:10]: Not that familiar with that syntax. Something like this should work, right?

```ruby
    it 'guesses by filename' do
      assert_guess :filename => 'foo.groovy'
      assert_guess :filename => 'Jenkinsfile'
      assert_guess :filename => 'foo.Jenkinsfile'
    end
```

Comment (tancnle) [reply lib/rouge/lexers/groovy.rb:10]: @danilaml Yep that will do 👍🏼

**Comment by tancnle** [comment]
Awesome. Thank you for the prompt action on this @danilaml. Let's ship it! 🚀

**Pull request merged**